### PR TITLE
Support for self-healing Consultation URL 

### LINKED ISSUE #6761: Support for self-healing Consultation URL and redirecting `/consultation/:id` to the actual route
**Is your feature request related to a problem? Please describe.**
Currently when we navigate to `/facility/<invalid_id>/patient/<invalid_id>/consultation/<valid_consultation_id>`, we still land on a working consultation details page, however two issues:

1. It's not a valid URL, auto-redirect to healed URL is not working.
2. Breadcrumbs, although show proper data, the `href` of each crumb points to an invalid URL.

**Describe the solution you'd like**

We can reconstruct the URL to point to a proper path since we can successfully retrieve a valid consultation record (since the consultation ID is/must be valid) and the object holds the valid patient and facility `id`.

- [ ] If the facility/patient ID is invalid, perform a redirect to the valid URL.
- [ ] Add a redirect route `care.app/consultation/<consultation_id>` that would redirect to the actual URL (`/facility/:id/patient/:id/consultation/:id`).

### PR DESCRIPTION
**Pull request by konavivekramakrishna**
## Proposed Changes

- Fixes #6761 
@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.coronasafe.network/coronasafe-care-documentation/architecture/architecture-and-layering-of-care).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### CODE REVIEW

File: src/Components/Facility/ConsultationDetails/index.tsx [hunk 1]
```diff
@@ -149,6 +147,14 @@ export const ConsultationDetails = (props: any) => {
                 : "No",
               is_vaccinated: patientData.is_vaccinated ? "Yes" : "No",
             };
+            if (
+              facilityId != data.facility_object?.id ||
+              patientId != data.id
```
Comment (rithviknishad) [inline src/Components/Facility/ConsultationDetails/index.tsx]: `data` here represents "Patient" object. Patient's facility need not always be the same as "Consultation" object's facility, as a patient can have multiple consultations, in different facilities. 

Hence this won't work as expected for an older consultation of a patient if the previous consultation belongs to a different facility than the current active consultation's facility right?

Do the check against `consultation.facility` instead of `patient (data).id`

File: src/Components/Facility/ConsultationDetails/index.tsx [hunk 2]
```diff
@@ -452,7 +458,11 @@ export const ConsultationDetails = (props: any) => {
                     <Link
                       key={p.text}
                       className={tabButtonClasses(tab === p.text)}
-                      href={`/facility/${facilityId}/patient/${patientId}/consultation/${consultationId}/${p.text.toLocaleLowerCase()}`}
+                      href={`/facility/${
+                        patientData?.facility_object?.id
+                      }/patient/${
+                        patientData?.id
+                      }/consultation/${consultationId}/${p.text.toLocaleLowerCase()}`}
```
Comment (rithviknishad) [inline src/Components/Facility/ConsultationDetails/index.tsx]: Why were these modified? Anyways the `facilityId`/`patientId` would end up being correct since it'll redirect if facility/patient ID is wrong right?

File: src/Components/Facility/ConsultationDetails/index.tsx [hunk 3]
```diff
@@ -466,15 +476,15 @@ export const ConsultationDetails = (props: any) => {
...
         show={showDoctors}
         setShow={setShowDoctors}
       />
 
       {showPatientNotesPopup && (
         <PatientNotesSlideover
-          patientId={patientId}
-          facilityId={facilityId}
+          patientId={String(patientData?.id)}
+          facilityId={String(patientData?.facility_object?.id)}
```
Comment (rithviknishad) [inline src/Components/Facility/ConsultationDetails/index.tsx]: Same, unnecessary modification

**Review by rithviknishad — CHANGES_REQUESTED** [review]

**Review by rithviknishad — APPROVED** [review]

**Comment by nihal467** [comment]
LGTM

**Review by khavinshankar — APPROVED** [review]

**Comment by rithviknishad** [comment]
lint checks has failed cc: @Ashesh3 

Screenshot 2023-12-29 at 22 36 18


Screenshot 2023-12-29 at 22 36 33
